File: arrays/1260_Shift2DGrid.py
from typing import List

class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        width, height = len(grid[0]), len(grid)
        for i in range(k):
            grid = self.shift(grid, width, height)
        return grid
    
    def shift(self, grid, width, height):
        shifted = [[0 for x in range(width)] for y in range(height)] 
        for row in range(height):
            for col in range(width):
                if row == height - 1 and col == width - 1:
                    shifted[0][0] = grid[row][col]
                elif col == width - 1:
                    shifted[row+1][0] = grid[row][col]
                else:
                    shifted[row][col+1] = grid[row][col]
        return shifted

solution = Solution()
print(solution.shiftGrid(grid = [[1,2,3],[4,5,6],[7,8,9]], k = 1))
# print(solution.shiftGrid(grid = [[3,8,1,9],[19,7,2,5],[4,6,11,10],[12,0,21,13]], k = 4))
# print(solution.shiftGrid([[1],[2],[3],[4],[7],[6],[5]], 23))
# print(solution.shiftGrid([[1,2,3,4,5,6]], 1))

File: arrays/1260_Shift2DGrid.cc
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int height = grid.size(), width = grid[0].size();
        for(int i = 0; i < k; i++){
            grid = this->shift(grid, width, height);
        }
        return grid;
    }

    vector<vector<int>> shift(vector<vector<int>>&src, int width, int height){
        vector<vector<int>> result(height, vector<int>(width));
        for(int i = 0; i < height; i++){
            for(int j = 0; j < width; j++){
                if(i == height-1 && j == width-1){
                    result[0][0] = src[i][j];
                }
                else if(j == width - 1){
                    result[i+1][0] = src[i][j];
                }
                else{
                    result[i][j+1] = src[i][j];
                }
            }
        }
        return result;
    }
};